internal/controllers/topology/cluster/structuredmerge/serversidepathhelper_test.go

```diff
@@ -506,3 +521,282 @@ func getTopologyManagedFields(original client.Object) map[string]interface{} {
 	}
 	return r
 }
+
+// NOTE: This test ensures that ServerSideApply works as expected when new defaulting logic is introduced by a Cluster API update.
+func TestServerSideApplyWithDefaulting(t *testing.T) {
+	g := NewWithT(t)
+
+	// Create a namespace for running the test
+	ns, err := env.CreateNamespace(ctx, "ssa-defaulting")
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Setup webhook with the manager.
+	// Note: The webhooks is not active yet, as the MutatingWebhookConfiguration will be deployed later.
+	mutatingWebhookConfiguration, err := setupWebhookWithManager(ns)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	// Calculate KubeadmConfigTemplate.
+	kct := &bootstrapv1.KubeadmConfigTemplate{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "kct",
+			Namespace: ns.Name,
+		},
+		Spec: bootstrapv1.KubeadmConfigTemplateSpec{
+			Template: bootstrapv1.KubeadmConfigTemplateResource{
+				Spec: bootstrapv1.KubeadmConfigSpec{
+					JoinConfiguration: &bootstrapv1.JoinConfiguration{
+						NodeRegistration: bootstrapv1.NodeRegistrationOptions{
+							KubeletExtraArgs: map[string]string{
+								"eviction-hard": "nodefs.available<0%,nodefs.inodesFree<0%,imagefs.available<0%",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// The test does the following.
+	// 1. Create KubeadmConfigTemplate
+	// 2. Activate the new defaulting logic via the webhook
+	//   * This simulates the deployment of a new Cluster API version with new defaulting
+	// 3. Simulate defaulting on original and/or modified
+	//   * defaultOriginal will add a label to the KubeadmConfigTemplate which will trigger defaulting
+	//     * original is the KubeadmConfigTemplate referenced in a MachineDeployment of the Cluster topology
+	//   * defaultModified will simulate that defaulting was run on the KubeadmConfigTemplate referenced in the ClusterClass
+	//     * modified is the desired state calculated based on the KubeadmConfigTemplate referenced in the ClusterClass
+	//   * We are testing through all permutations as we don't want to assume on which objects defaulting was run.
+	// 4. Check patch helper results
+
+	// We have the following test cases:
+	// | original   | modified    | expect behavior                                           |
+	// |            |             | no-op                                                     |
+	// | defaulted  |             | no-op                                                     |
+	// |            | defaulted   | no spec changes, only take ownership of defaulted fields  |
+	// | defaulted  | defaulted   | no spec changes, only take ownership of defaulted fields  |
+	tests := []struct {
+		name                 string
+		defaultOriginal      bool
+		defaultModified      bool
+		expectChanges        bool
+		expectSpecChanges    bool
+		expectFieldOwnership bool
```

Comment:
nt (feel free to ignore)
expectSpecChanges is always false, we can drop it. 
expectFieldOwnership is the same of expectChanges, we can drop it too

Reply:
I would like to keep it as is. The idea is to make it very obvious in the table what the behavior is